tests/test_datasets/test_dataset.py

```diff
@@ -956,20 +957,45 @@ def test_train_test_split_pass_regressors_to_output(df_and_regressors):
     df, df_exog, known_future = df_and_regressors
     ts = TSDataset(df=df, df_exog=df_exog, freq="D", known_future=known_future)
     train, test = ts.train_test_split(test_size=5)
+    assert set(train.regressors).issubset(set(train.features))
+    assert set(test.regressors).issubset(set(test.features))
+    assert train.regressors == ts.regressors
+    assert test.regressors == ts.regressors
+
+
+def test_train_test_split_pass_transform_regressors_to_output(df_and_regressors):
+    df, df_exog, known_future = df_and_regressors
+    ts = TSDataset(df=df, df_exog=df_exog, freq="D", known_future=known_future)
+    ts.fit_transform(transforms=[LagTransform(in_column="target", lags=[1, 2, 3])])
+    train, test = ts.train_test_split(test_size=5)
+    assert set(train.regressors).issubset(set(train.features))
+    assert set(test.regressors).issubset(set(test.features))
     assert train.regressors == ts.regressors
     assert test.regressors == ts.regressors
 
 
 def test_train_test_split_pass_target_components_to_output(ts_with_target_components):
     train, test = ts_with_target_components.train_test_split(test_size=5)
+    train_target_components = train.get_target_components()
+    test_target_components = test.get_target_components()
+    assert set(train.target_components_names).issubset(set(train.features))
+    assert set(test.target_components_names).issubset(set(test.features))
     assert sorted(train.target_components_names) == sorted(ts_with_target_components.target_components_names)
     assert sorted(test.target_components_names) == sorted(ts_with_target_components.target_components_names)
+    assert set(train_target_components.columns.get_level_values("feature")) == set(train.target_components_names)
+    assert set(test_target_components.columns.get_level_values("feature")) == set(test.target_components_names)
 
 
 def test_train_test_split_pass_prediction_intervals_to_output(ts_with_prediction_intervals):
```

Comment:
It would be nice to add a similar test for hierarchy, testing that structure is copied and current levels are preserved. Seems like we lack this test in general.